KsApi/extensions/NSURLSession.swift

```diff
@@ -12,6 +12,65 @@ private func parseJSONData(_ data: Data) -> Any? {
 private let scheduler = QueueScheduler(qos: .background, name: "com.kickstarter.ksapi", targeting: nil)
 
 internal extension URLSession {
+  // Wrap an URLSession producer with Graph error envelope logic.
+  func rac_graphDataResponse(_ request: URLRequest)
+    -> SignalProducer<Data, GraphError> {
+    let producer = self.reactive.data(with: request)
+
+    return producer
+      .start(on: scheduler)
+      .flatMapError { error -> SignalProducer<(Data, URLResponse), GraphError> in
+        print("🔴 [KsApi] Request Error \(error.localizedDescription)")
+
+        return .init(error: GraphError.requestError(error, nil))
+      }
+      .flatMap(.concat) { data, response -> SignalProducer<Data, GraphError> in
+        guard let response = response as? HTTPURLResponse else { fatalError() }
+
+        guard
+          (200..<300).contains(response.statusCode),
+          let headers = response.allHeaderFields as? [String: String],
+          let contentType = headers["Content-Type"], contentType.hasPrefix("application/json")
+        else {
+          print("🔴 [KsApi] HTTP Failure \(self.sanitized(request))")
```

Comment:
Seems to be an extra parenthesis at the end of this string.

Reply:
I think actually it has the correct number of parenthesis 🤔